salt/modules/virt.py

```diff
@@ -353,6 +353,108 @@ def _qemu_image_info(path):
     return ret
 
 
+def _qemu_image_create(vm_name,
+                       disk_file_name,
+                       disk_size,
+                       disk_image,
+                       disk_type='qcow2',
+                       enable_qcow=False,
+                       saltenv='base'):
+    '''
+    Create the image file using specified disk_size or/and disk_image
+
+    Return path to the created image file
+    '''
+    img_dir = __salt__['config.option']('virt.images')
+    log.debug('Image directory from config option `virt.images`'
+              ' is {0}'.format(img_dir))
+    img_dest = os.path.join(
+        img_dir,
+        vm_name,
+        disk_file_name
+    )
+    log.debug('Image destination will be {0}'.format(img_dest))
+    img_dir = os.path.dirname(img_dest)
+    log.debug('Image destination directory is {0}'
+              .format(img_dir))
+    try:
+        os.makedirs(img_dir)
+    except OSError:
+        pass
+
+    if disk_image:
+        log.debug('Create disk from specified image {0}'
+                  .format(disk_image))
+        sfn = __salt__['cp.cache_file'](disk_image, saltenv)
+
+        qcow2 = False
+        if salt.utils.which('qemu-img'):
+            res = __salt__['cmd.run']('qemu-img info {}'.format(sfn))
+            imageinfo = yaml.load(res)
+            qcow2 = imageinfo['file format'] == 'qcow2'
+        try:
+            if enable_qcow and qcow2:
+                log.info('Cloning qcow2 image {0} using copy on write'
+                         .format(sfn))
+                __salt__['cmd.run'](
+                    'qemu-img create -f qcow2 -o backing_file={0} {1}'
+                    .format(sfn, img_dest).split())
+            else:
+                log.debug('Copying {0} to {1}'.format(sfn, img_dest))
+                salt.utils.files.copyfile(sfn, img_dest)
+
+            mask = os.umask(0)
+            os.umask(mask)
+
+            if disk_size and qcow2:
```

Comment:
Just don't set 'size' attribute in the disk profile to avoid resizing, and it will be initialized with None in L744.

For standalone use of this method your comment makes sense, thanks!